src/Uno.UI/UI/Xaml/UIElement.Pointers.cs

```diff
@@ -268,7 +273,34 @@ internal struct PointerEventDispatchResult
 		internal HitTestability GetHitTestVisibility()
 		{
 #if __WASM__ || __SKIA__ || __MACOS__
-			return HitTestVisibility;
+			var element = this;
+			while (element is not null)
+			{
+				if (!element.IsEnabledOverride() || !element.IsHitTestVisible)
+				{
+					return HitTestability.Collapsed;
+				}
+
+				element = GetParentUIElement(element);
```

Comment:
I don't see any interest to walk the tree up instead of relying on a pre-computed property (that is pushed to native elements on wasm for instance). This will be perform for each pointer so I would assume this change would have a negative perf impact.

Reply:
It should not: the down propagation is computed only when an element is being added into the tree or something that changes testability is updated... 

Even if this approach is closer to winUI, we need to pre-compute the testability for native elements, so I don't see any interest to try to change the behavior.